src/pages/index.js

```diff
@@ -2,85 +2,223 @@ import "./index.css";
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "6a75e942-1cdf-4570-be93-f1316f67106a",
+    "Content-Type": "application/json",
+  },
+});
+
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
+}
 
-const previewImageModal = new PopupWithImage("#preview-image-modal");
-previewImageModal.setEventListeners();
-const userInfo = new UserInfo(".profile__title", ".profile__description");
 const itemSection = new Section({
-  items: initialCards,
-  renderer: renderCard,
+  items: [],
   cssSelector: ".cards__list",
+  renderer: renderCard,
 });
-itemSection.renderItems();
 
-const addCardModal = new PopupWithForm(
-  "#profile-add-card",
-  handleCardAddSubmit
+api
+  .getCardList()
+  .then((items) => {
+    itemSection.renderItems(items);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
-
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
-}
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.setUserInfo(data);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
-}
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      deleteConfirm.renderDelete(true);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      deleteConfirm.renderDelete(false);
+      deleteConfirm.close();
+    });
+
+const handleLikeClick = (card) => {
+  api
+    .toggleCardLike(card._id, card._isLiked)
+    .then((cardData) => {
+      card.getLikeStatus(cardData.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
+
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  function handleChangePicture(avatar) {
+    changePictureModal.renderLoading(true, "saving...", "save");
+    api
+      .setUserAvatar(avatar)
+      .then((data) => {
+        userInfo.setUserAvatar(data.avatar);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        changePictureModal.renderLoading(false, "saving...", "save");
+        changePictureModal.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

Reply:
* You haven't corrected my remark https://github.com/MrWatson0531/se_project_aroundtheus/pull/1/files#r1606096636   You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

`changePictureModal.close();` should be in line 139 in block `then`